ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,201 @@ def test_ipahealthcheck_trust_agent_member_check(self):
+        )
+
+    def test_ipahealthcheck_ipaopensslchainvalidation(self, rename_httpd_cert):
+        """
+        Test for IPAOpenSSLChainValidation
+        """
+        error_msg = "Can't open {} for reading".format(paths.HTTPD_CERT_FILE)
+        verification_msg = "Certificate validation for {} failed".format(
+            paths.HTTPD_CERT_FILE
+        )
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
+                assert check["result"] == "ERROR"
+                assert error_msg in check["kw"]["reason"]
+                assert verification_msg in check["kw"]["msg"]
+
+    @pytest.fixture
+    def remove_server_cert(self):
+        """
+        Fixture to remove Server cert and revert the change.
+        """
+        instance = realm_to_serverid(self.master.domain.realm)
+        self.master.run_command(
+            [
+                "certutil",
+                "-L",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-n",
+                "Server-Cert",
+                "-a",
+                "-o",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+                + "/Server-Cert.pem",
+            ]
+        )
+        self.master.run_command(
+            [
+                "certutil",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-D",
+                "-n",
+                "Server-Cert",
+            ]
+        )
+        yield
+        self.master.run_command(
+            [
+                "certutil",
+                "-d",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance,
+                "-A",
+                "-i",
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+                + "/Server-Cert.pem",
+                "-t",
+                "u,u,u",
+                "-f",
+                paths.IPA_NSSDB_PWDFILE_TXT,
+                "-n",
+                "Server-Cert",
+            ]
+        )
+
+    def test_ipahealthcheck_ipansschainvalidation(self, remove_server_cert):
+        """
+        Test for IPANSSChainValidation
+        """
+        error_msg = (
+            ': certutil: could not find certificate named "Server-Cert": '
+            "PR_FILE_NOT_FOUND_ERROR: File not found\n"
+        )
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPANSSChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["nickname"] == "Server-Cert":
+                assert check["result"] == "ERROR"
+                assert check["kw"]["reason"] == error_msg
```

Comment:
This is ok I guess but without Server-Cert the service won't start at all so so it would be pretty obvious. I think you delete the CA cert before this that would better exercise things. You could run ipa-certupdate afterward, I think, to re-add the CA cert.